From a5b586a3a6a244725d8cf6e92388c10fd4b70c55 Mon Sep 17 00:00:00 2001
From: Don Darling <ddarling@ti.com>
Date: Wed, 13 Oct 2010 18:20:08 -0500
Subject: [PATCH 15/16] v4l2src: Remove short-circuit of VIDIOC_S_FMT call

If VIDIOC_G_FMT returns the format we need, v4l2src will skip the call to
VIDIOC_S_FMT because it is not needed.  However, on DM365 a call to
VIDIOC_G_FMT will return that it is set to 1280x720 UYVY when it really isn't
causing UYVY capture to behave incorrectly.  This change removes that
optimization and always calls VIDIOC_S_FMT.
---
 sys/v4l2/gstv4l2object.c |   16 ----------------
 1 files changed, 0 insertions(+), 16 deletions(-)

diff --git a/sys/v4l2/gstv4l2object.c b/sys/v4l2/gstv4l2object.c
index 2d79b0b..e63a762 100644
--- a/sys/v4l2/gstv4l2object.c
+++ b/sys/v4l2/gstv4l2object.c
@@ -1987,22 +1987,6 @@ gst_v4l2_object_set_format (GstV4l2Object * v4l2object, guint32 pixelformat,
   if (v4l2_ioctl (fd, VIDIOC_G_FMT, &format) < 0)
     goto get_fmt_failed;
 
-  if (format.type == v4l2object->type &&
-      format.fmt.pix.width == width &&
-      format.fmt.pix.height == height &&
-      format.fmt.pix.pixelformat == pixelformat) {
-    /* Nothing to do. We want to succeed immediately
-     * here because setting the same format back
-     * can still fail due to EBUSY. By short-circuiting
-     * here, we allow pausing and re-playing pipelines
-     * with changed caps, as long as the changed caps
-     * do not change the webcam's format. Otherwise,
-     * any caps change would require us to go to NULL
-     * state to close the device and set format.
-     */
-    return TRUE;
-  }
-
   format.type = v4l2object->type;
   format.fmt.pix.width = width;
   format.fmt.pix.height = height;
-- 
1.7.0.4

